### ISSUE TITLE
`source.fixAll` (Fix All command) removes unused imports despite ignoring `reportUnusedImport`

### ISSUE THREAD

**Issue by Avasam** [microsoft/pylance-release#3091]
## Environment data

-   Language Server version: v2022.7.41
-   OS and version: Windows 10
-   Python version (& distribution if applicable, e.g. Anaconda): 3.9.13

## Code Snippet

```python
import sys  # pyright: ignore [reportUnusedImport]
```

## Repro Steps

1. In .vscode/settings.json, configure 
```json
"editor.codeActionsOnSave": {
  "source.fixAll": true
}
```

## Expected behavior

Don't remove the unused import. Or offer a way to configure not removing unused imports without having to completely disable `source.fixAll`

## Actual behavior

Import is removed even though I specifically ignored the `reportUnusedImport` error on that line.  
Same if I configure `[tool.pyright] reportUnusedImport = "none"` in `pyproject.toml`.

I've uninstalled all formatters and tried with Pylance enabled and disabled to make sure this extension is what's causing this behaviour.

## Logs

```
> "C:\Program Files\Python39\python.exe" -m autopep8 --max-line-length=120 --diff .\src\menu_bar.py.00c7022eb47e6bcec783281b333c04f2.tmp
cwd: .
> "C:\Program Files\Python39\python.exe" -c "import autopep8"
> "C:\Program Files\Python39\python.exe" ~\.vscode\extensions\ms-python.python-2022.11.12031009\pythonFiles\linter.py -m flake8 .\src\menu_bar.py
cwd: .
> "C:\Program Files\Python39\python.exe" ~\.vscode\extensions\ms-python.python-2022.11.12031009\pythonFiles\linter.py -m bandit .\src\menu_bar.py
cwd: .
> "C:\Program Files\Python39\python.exe" ~\.vscode\extensions\ms-python.python-2022.11.12031009\pythonFiles\linter.py -m pylint .\src\menu_bar.py
cwd: .
> "C:\Program Files\Python39\python.exe" -c "import autopep8"

Formatting with autopep8 failed.
You could either install the 'autopep8' formatter, turn it off or use another formatter.
Error: Module 'autopep8' not installed.
> "C:\Program Files\Python39\python.exe" -c "import flake8"
> "C:\Program Files\Python39\python.exe" -c "import bandit"
> "C:\Program Files\Python39\python.exe" -c "import pylint"
> "C:\Program Files\Python39\python.exe" -c "import flake8"
> "C:\Program Files\Python39\python.exe" -c "import bandit"
> "C:\Program Files\Python39\python.exe" -c "import pylint"

Linter 'flake8' is not installed. Please install it or select another linter".
Error: Module 'flake8' not installed.

Linter 'bandit' is not installed. Please install it or select another linter".
Error: Module 'bandit' not installed.

Linter 'pylint' is not installed. Please install it or select another linter".
Error: Module 'pylint' not installed.```

**Comment by erictraut** [comment]
Pylance doesn't expose a "fixAll" command. It must be one of the other linters that you have installed.

**Comment by Avasam** [comment]
Could it be from `ms-python.python` then?
Pylance Disabled:
![image](https://user-images.githubusercontent.com/1350584/180630417-fe6e02eb-cfec-4730-b971-58106dd2bf9a.png)

Pylance Enabled:
![image](https://user-images.githubusercontent.com/1350584/180630438-6c797199-5365-408a-b48f-15c6fbedcb77.png)

**Comment by heejaechang** [comment]
Pylance recently added fix all for unused import. let me take a look and make sure we don't remove unused imports whose diagnostic is disabled.

**Comment by heejaechang** [comment]
linking related feature working on by vscode - https://github.com/microsoft/vscode/issues/82718

**Comment by heejaechang** [comment]
@Avasam in ```next prerelease```, unused import ```fix all code action``` will have its own name  ```source.fixAll.unusedImports```, you should be able to use that to configure your editor.codeActionsOnSave.

For now, please use stable version as a workaround. here is our changelog that shows what's in the prerelease you are using. (https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2022741-20-july-2022)

**Comment by heejaechang** [comment]
This issue has been fixed in version 2022.7.42, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2022742-27-July-2022)

**Comment by heejaechang** [comment]
forgot to mention. this contains all source action we provide

https://github.com/microsoft/pylance-release/blob/main/README.md#source-code-actions
